pkg/master/services.go

```diff
@@ -35,12 +37,18 @@ func ServiceIPRange(passedServiceClusterIPRange net.IPNet) (net.IPNet, net.IP, e
 		klog.Warningf("No CIDR for service cluster IPs specified. Default value which was %s is deprecated and will be removed in future releases. Please specify it using --service-cluster-ip-range on kube-apiserver.", kubeoptions.DefaultServiceIPCIDR.String())
 		serviceClusterIPRange = kubeoptions.DefaultServiceIPCIDR
 	}
-	if size := ipallocator.RangeSize(&serviceClusterIPRange); size < 8 {
+
+	size := integer.Int64Min(utilnet.RangeSize(&serviceClusterIPRange), 1<<16)
+	ones, bits := serviceClusterIPRange.Mask.Size()
+	if bits == 128 && (bits-ones) >= 16 {
```

Comment:
doesn't the Int64Min above remove the need to do this check?

Reply:
the `bits == 128 && (bits-ones) >= 127` case that returns 0 would fail the `< 8` check below (matching previous behavior)

the `bits == 128 && (bits-ones) >= 16` case would return at most MaxInt64, which would get chopped down to `1<<16` by the Int64Min call above (matching previous behavior)